kong/plugins/response-transformer/handler.lua

```diff
@@ -28,10 +29,11 @@ function ResponseTransformerHandler:body_filter(conf)
   if is_body_transform_set(conf) and is_json_body(ngx.header["content-type"]) then
     local chunk, eof = ngx.arg[1], ngx.arg[2]
     if eof then
-      local body = body_filter.transform_json_body(conf, ngx.ctx.buffer)
+      local body = body_filter.transform_json_body(conf, table.concat(ngx.ctx.chunks))
       ngx.arg[1] = body
     else
-      ngx.ctx.buffer = ngx.ctx.buffer .. chunk
+      ngx.ctx.chunks[ngx.ctx.chunk_number] = chunk
+      ngx.ctx.chunk_number = ngx.ctx.chunk_number + 1
```

Comment:
in this code there are a lot of `ngx.ctx` accesses. Since the lookup of `ctx` in `ngx` is rather expensive (it has some meta table magic), it would be better to cache it in a local.

```lua
-- at the top
local ctx = ngx.ctx`
```

and then replace all `ngx.ctx.whatever` to `ctx.whatever`

Reply:
context cached in local variable in both ('access' and 'body_filter') functions